### PR TITLE
Gz tree

### PR DESCRIPTION
**Pull request by memanuel**
Changes made to support training Galaxy Zoo data set.

datasets.py: for GalaxyZoo data set, need to manually split training samples into train (90%) and validation (10%). This is because the test set for GalaxyZoo doesn't include ground truth labels.

genotypes.py: added the most recent architecture for GalaxyZoo found 2019-11-05.

model.py: created a new NetworkGalaxyZoo class.  This is analogous to the NetworkGalaxyZoo class  model_search.py that extends Network in model_search.py.  It contains two fully connected layers and the GalaxyZoo decision tree.

model_search.py: Changed Cell class so primitives is no longer a global constant, but an input argument passed by string name. This allows us to run architecture searches on different datasets with different sets of primitives.  This change was necessary to support including the 2x2 max pool operation in the architecture search (2x2 max pool was used multiple times in the winning entry.)  This change also rippled through to Network and NetworkGalaxyZoo.  Primitives are now stored on a table keyed by primitive_name (a string) with value a list of operation names.  The NetworkGalaxyZoo class has a similar interface to Network, with fully connected layers and the GZ decision tree at the end.

operations.py: Added max_pool_2x2 operation.  This required asymmetric padding so the size would remain the same.

train.py: Added new command line arguments to support GalaxyZoo.  These can be ignored when not training GZ.  Architecture will now default to a lookup table keyed by dataset name.  Can still manually name the architecture with the --arch flag the old way.  Added Adam optimizer; this is now the default with a LR of 1.0E-3.  Logging now includes the MSE (R2 is incorrect for GZ, so this is essential to monitor progress in training.)  Training results are saved to numpy arrays so we don't need to strip log files.  (This is also true for train_search.py)

train_search.py: Added NetworkGalaxyZoo class.  Extends Network API with FC layers and GZ decision tree.  Changed default weight decay to 1.0E-6 b/c 3.0E-4 looked very high.  Added Adam optimizer. This is now the default with LR=1.0E-3.  Can specify desired primities with --primitives argument.

utils.py: Added data transforms for

### CODE REVIEW

**Comment by dylanrandle** [comment]
Looks great to me. fixes the issue with validation data in `train.py` as well, which is a nice bonus!
